migration/migration.go

```diff
@@ -607,22 +608,26 @@ func createOrUpdateSpace(ctx context.Context, spaceRepo *space.GormRepository, i
 }
 
 func createSpace(ctx context.Context, spaceRepo *space.GormRepository, id uuid.UUID, description string) error {
-	_, err := spaceRepo.Load(ctx, id)
-	cause := errs.Cause(err)
-	newSpace := &space.Space{
-		Description: description,
-		Name:        "system.space",
-		ID:          id,
-	}
-	switch cause.(type) {
-	case errors.NotFoundError:
-		log.Info(ctx, map[string]interface{}{
-			"pkg":      "migration",
-			"space_id": id,
-		}, "space %s will be created", id)
-		_, err := spaceRepo.Create(ctx, newSpace)
-		if err != nil {
-			return errs.Wrapf(err, "failed to create space %s", id)
+	exists, err := spaceRepo.Exists(ctx, id.String())
+	if !exists {
+		cause := errs.Cause(err)
+		newSpace := &space.Space{
```

Comment:
I think this `newSpace` can be moved inside the `case errors.NotFoundError` block.